graph/src/data/subgraph/mod.rs

```diff
@@ -588,18 +605,17 @@ impl UnresolvedMapping {
 
         info!(logger, "Resolve mapping"; "link" => &link.link);
 
-        // resolve each abi
-        stream::futures_ordered(
-            abis.into_iter()
-                .map(|unresolved_abi| unresolved_abi.resolve(resolver, logger.clone())),
-        )
-        .collect()
-        .join(
-            resolver.cat(&logger, &link).and_then(|module_bytes| {
-                Ok(Arc::new(parity_wasm::deserialize_buffer(&module_bytes)?))
-            }),
+        let (abis, runtime) = try_join(
+            // resolve each abi
+            collect_futures(
+                abis.into_iter()
+                    .map(|unresolved_abi| unresolved_abi.resolve(resolver, logger)),
+            ),
```

Comment:
I _may_ be wrong, but we might be able to use unordered futures here to resolve all these files in parallel. I don't think the order of the abis in `Mapping` matters, since they are identified by their names anyway (e.g. when the mapping makes contract calls).

Reply:
Using `FuturesOrdered` does not restrict the parallelism for retrieving the files. Each file is retrieved concurrently, and `FuturesOrdered` buffers any results that are available but not yet ready to be yielded. So, while there is a performance benefit to choose `FuturesUnordered`, it's not that big.

So, while you're probably right, I'm a bit hesitant to stir the pot more on this change for such a small gain. If we intend to make this an unordered collection I'd prefer it to be a HashSet or something.

Reply:
Oh, you're absolutely right, they are still resolved in parallel... this is fine then.